src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs

```diff
@@ -218,13 +242,14 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
             region_sub
         );
 
-         match find_use::find(mir, regioncx, tcx, region_sub, context.loc) {
+        match find_use::find(mir, regioncx, tcx, region_sub, context.loc) {
             Some(Cause::LiveVar(local, location)) => {
                 let span = mir.source_info(location).span;
-                let spans = self.move_spans(&Place::Local(local), location)
+                let spans = self
+                    .move_spans(&Place::Local(local), location)
                     .or_else(|| self.borrow_spans(span, location));
 
-                if self.is_borrow_location_in_loop(context.loc) {
+                if self.can_reach_ignoring_backedges(context.loc, location) {
```

Comment:
it should be `if !self.can_reach_ignoring_backedges(...) { .. }` -- basically, if you can reach it without traversing a backedge, then you **don't** need to use the loop to explain how one reaches the other. But if you **must** traverse a backedge, you do.